### PR TITLE
Lock conda dependency versions

### PR DESCRIPTION
**Pull request by weiji14**
Ensure reproducibility of the software environment. Created using `mamba list --explicit > environment-linux-64.lock`.

We actually have a pip-only dependency `lazrs`, so I might try using [`conda-lock`](https://github.com/conda-incubator/conda-lock/tree/v1.0.5)'s new toml lockfile format to get it 100% reproducible.

Note: Might need to uninstall `lazrs` and clear poetry's cache according to https://github.com/python-poetry/poetry/issues/4679 in order to prevent `RuntimeError: Unable to find installation candidates for lazrs (0.4.0)`. Specifically, do:

```
pip uninstall lazrs
poetry cache clear --all pypi
rm -rf ~/.cache/pypoetry/
rm -rf ~/.cache/pip/
conda-lock -f environment.yml
```

~~Edit: Still not working because `lazrs` doesn't have an sdist :sweat_smile:, Ref: https://github.com/python-poetry/poetry/discussions/5574~~ Edit2: lazrs==0.4.3 now has an sdist so conda-lock works, ref: https://github.com/laz-rs/laz-rs-python/issues/13

### CODE REVIEW

File: conda-lock.yml [hunk 1]
```diff
@@ -0,0 +1,4591 @@
+# This lock file was generated by conda-lock (https://github.com/conda-incubator/conda-lock). DO NOT EDIT!
+#
+# A "lock file" contains a concrete list of package versions (with checksums) to be installed. Unlike
+# e.g. `conda env create`, the resulting environment will not change as new package versions become
+# available, unless you explicitly update the lock file.
+#
+# Install this environment as "YOURENV" with:
+#     conda-lock install -n YOURENV --file conda-lock.yml
```
Comment (weiji14) [inline conda-lock.yml:8]: Official conda-lock instructions seems to be wrong here, had to remove the `--file`
```suggestion
#     conda-lock install -n YOURENV conda-lock.yml
```
Or for this repo, do `conda-lock install -n egu22pygmt conda-lock.yml`

File: conda-lock.yml [hunk 2]
```diff
@@ -0,0 +1,4591 @@
...
+metadata:
+  channels:
+  - url: conda-forge
+    used_env_vars: []
+  - url: nodefaults
+    used_env_vars: []
+  content_hash:
+    linux-64: 27752ab8a15c7be522d4cabfe3ae7906d7ea7cabae18937ecebf74ea2dad3612
+  platforms:
+  - linux-64
```
Comment (weiji14) [inline conda-lock.yml:26]: Debating on whether to have a Linux only lockfile, or do macOS and Windows as well :upside_down_face:

Comment (maxrjones) [reply conda-lock.yml:26]: TBH I think linux only is fine since that's what we used to deploy the book.

Comment (weiji14) [reply conda-lock.yml:26]: On second thought, let's just lock it. It adds like 9000+ lines to the `conda-lock.yml` file but at least macOS and Windows users can reproduce the environment fully too.

File: conda-lock.yml (hunk 1 above)
Comment (weiji14) [inline conda-lock.yml]: Ok, this seems to be the safer way to do things (though not very intuitive).
```suggestion
#     conda create --name egu22pygmt conda-lock=1.0.5
#     conda activate egu22pygmt
#     conda-lock install --name egu22pygmt conda-lock.yml
```

File: conda-lock.yml [hunk 3]
```diff
@@ -0,0 +1,13147 @@
+# This lock file was generated by conda-lock (https://github.com/conda-incubator/conda-lock). DO NOT EDIT!
+#
+# A "lock file" contains a concrete list of package versions (with checksums) to be installed. Unlike
+# e.g. `conda env create`, the resulting environment will not change as new package versions become
+# available, unless you explicitly update the lock file.
+#
+# Install this environment as "YOURENV" with:
+#     conda-lock install -n YOURENV conda-lock.yml
```
Comment (weiji14) [inline conda-lock.yml]: Ok, this seems to be the safer way to do things (though not very intuitive).
```suggestion
#     conda create --name egu22pygmt conda-lock=1.0.5
#     conda activate egu22pygmt
#     conda-lock install --name egu22pygmt conda-lock.yml
```

**Pull request merged**